cmd/csi-node-driver-registrar/node_register.go

```diff
@@ -60,8 +62,25 @@ func nodeRegister(csiDriverName, httpEndpoint string) {
 	}
 	klog.Infof("Registration Server started at: %s\n", socketPath)
 	grpcServer := grpc.NewServer()
+
 	// Registers kubelet plugin watcher api.
-	registerapi.RegisterRegistrationServer(grpcServer, registrar)
+	if runtime.GOOS == "windows" {
+		for true {
+			// try register on Windows node in a loop, detailed issue:
+			// https://github.com/kubernetes-csi/node-driver-registrar/issues/143
+			err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+				registerapi.RegisterRegistrationServer(grpcServer, registrar)
```

Comment:
The issue description says that this call hangs forever. I don't think this Poll method is going to ever return in that case. I think you'll need to run the call in a go routine and wait for the result or timeout that way.

Reply:
I think that's a wrong fix, do you know is there any way to restart node-driver-registrar automatically if register is not succeeded with a few seconds? The registration issue happens on both Azure & GKE Windows node now. When we hit the issue, restart node driver always works.

Reply:
I think returning a fatal error is the main way to cause node-driver-registrar to restart.

Reply:
From the log you showed that the hang happened after this registration process. So will loop here help?

Reply:
This PR won't work, after `registerapi.RegisterRegistrationServer`, it just wait for following `GetInfo` invoked, while sometimes on Windows node, it never happens.

https://github.com/kubernetes-csi/node-driver-registrar/blob/1c2d6062e2b2d46b43e2c1d214652f2b50f14446/cmd/csi-node-driver-registrar/main.go#L78-L80